### PR TITLE
Dev/vrml output

### PR DESCRIPTION
**Pull request by theweaklink**
Adding 3D scene through new `PyvistaVrmlOutput` element

Screenshot from 2023-09-14 15-35-14

### CODE REVIEW

File: examples/PyVistaViz/data/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+## Info
+
+Place here your data. It is the default directory where getting/putting data unless specified otherwise
+(see onecode.Project()).
```
Comment (hameye) [inline examples/PyVistaViz/data/README.md:4]: Remove new line.

File: examples/PyVistaViz/flows/onecode_ext/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+## OneCode Extensions
+
+Optionally put your input/output elements in this extension.
+
+#### Usage
+Simply import `onecode_ext`, e.g.
+
+```python
+# ./flows/my_flow.py
```
Comment (hameye) [inline examples/PyVistaViz/flows/onecode_ext/README.md:9]: Should this be here ?

File: onecode/elements/output/plotly_output.py [hunk 3]
```diff
@@ -90,7 +105,7 @@ def _validate(
             ValueError: if the file does not have a JSON extension `.json`.
 
         """
-        _, ext = os.path.splitext(self.value)
+        _, ext = os.path.splitext(value)
```
Comment (hameye) [inline onecode/elements/output/plotly_output.py:108]: What is the difference between self.value and value ?

File: examples/PyVistaViz/.onecode.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+[
+    {
+        "file": "pyvistaviz",
+        "label": "PyVistaViz",
+        "attributes": {}
+    }
+]
```
[the remark this answers is not in the document]
Comment (theweaklink) [reply examples/PyVistaViz/.onecode.json:7]: :+1:

File: examples/PyVistaViz/data/README.md (hunk 1 above)
Comment (theweaklink) [reply examples/PyVistaViz/data/README.md:4]: will do but for `md` we can split among new lines for readability without immpacting the actual formatting

File: examples/PyVistaViz/.onecode.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (theweaklink) [reply examples/PyVistaViz/.onecode.json:7]: note that this file is auto-generated by the CLI
```
with open(config_file, 'w') as f:
        json.dump(flows, f, indent=4)
```
so we don't control the EOF

File: examples/PyVistaViz/flows/onecode_ext/README.md (hunk 2 above)
Comment (theweaklink) [reply examples/PyVistaViz/flows/onecode_ext/README.md:9]: This is automatically generated by the CLI `onecode-create` so we can debate the skeleton in another PR?
Skeleton is here: https://github.com/deeplime-io/onecode/tree/main/onecode/cli/skeleton

File: onecode/elements/output/plotly_output.py (hunk 3 above)
Comment (theweaklink) [reply onecode/elements/output/plotly_output.py:108]: In the case of this element, no difference, as the code calling validate is:
```py
        val = self.value
        self._validate(val)
```

however there could be cases with need to validate on the specified value and not the internal one (especially if the element is extended).
So using `self._value` is potentially bug prone

**Review by hameye — APPROVED** [review]

**Pull request merged**